integrate.js

```diff
@@ -0,0 +1,172 @@
+ *    list of conditions and the following disclaimer. 
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution. 
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+"use strict";
+
+(function(Nuvola)
+{
+
+// Create media player component
+var player = Nuvola.$object(Nuvola.MediaPlayer);
+
+// Handy aliases
+var PlaybackState = Nuvola.PlaybackState;
+var PlayerAction = Nuvola.PlayerAction;
+
+// Create new WebApp prototype
+var WebApp = Nuvola.$WebApp();
+
+// Initialization routines
+WebApp._onInitWebWorker = function(emitter)
+{
+    Nuvola.WebApp._onInitWebWorker.call(this, emitter);
+
+    this.state = PlaybackState.UNKNOWN;
+
+    document.addEventListener("DOMContentLoaded", this._onPageReady.bind(this));
+}
+
+// Page is ready for magic
+WebApp._onPageReady = function()
+{
+    // TODO: only set this after the user has logged in
+    // Connect handler for signal ActionActivated
+    Nuvola.actions.connect("ActionActivated", this);
+
+    // Start update routine
+    this.update();
+}
+
+WebApp.getMP3Player = function()
+{
+    var mp3Player;
+    try {
+        mp3Player = document.getElementById("mp3Player");
+    } catch (e) {
+        mp3Player = document;
+    }
+    return mp3Player;
+}
+
+// Extract data from the web page
+WebApp.update = function()
+{
+    var track = {
+        title: null,
+        artist: null,
+        album: null,
+        artLocation: null
+    }
+
+    var playerRoot = this.getMP3Player();
+
+    try {
+        var songDetails = playerRoot.getElementsByClassName("currentSongDetails")[0];
+        track.title = songDetails.getElementsByClassName("title")[0].innerText;
+        track.artist = songDetails.getElementsByClassName("artistLink")[0].innerText;
+
+        var albumImage = playerRoot.getElementsByClassName("albumImage")[0];
+        track.artLocation = albumImage.src;
+        if (track.album === null)
+            track.album = albumImage.title;
+    } catch (e) {
+        console.log("Failed to get track info");
+        console.log(e.message);
```

Comment:
These logging commands are pretty annoying. Could you comment them out with Geany-style toggle comments "//~"?